clippy_lints/src/methods/iter_on_single_or_empty_collections.rs

```diff
@@ -25,7 +29,25 @@ impl IterType {
     }
 }
 
-pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>, method_name: &str, recv: &Expr<'_>) {
+fn is_arg_ty_unified_in_fn<'tcx>(
+    cx: &LateContext<'tcx>,
+    fn_id: DefId,
+    arg_id: HirId,
+    args: impl Iterator<Item = &'tcx Expr<'tcx>> + Clone,
+) -> bool {
+    let fn_sig = cx.tcx.fn_sig(fn_id).instantiate_identity();
```

Comment:
Similarily `fn_sig` can ICE when called on something that's not a fn, like a `const` (this is also why the `ice-11065` test starts ICEing again).
```rs
fn foo(_: std::slice::Iter<()>) {}
const C: fn(std::slice::Iter<()>) = foo;
C([].iter());
```
likely needs a check `tcx.def_kind(fn_id).is_fn_like()` or some such

Reply:
I ended up skipping this check in general for calls to closures & callable constants, closure calls are always linted, callable constants calls are never linted, since usually closures are local variables, the signatures of which u can thus control, whereas constants are usually outside of your control (and frankly, callable constants are basically unicorns in regards to their rarity)